Applications/spire/source/common/local_technicals_model.cpp

```diff
@@ -7,54 +7,38 @@ using namespace Spire;
 
 LocalTechnicalsModel::LocalTechnicalsModel(Security security)
     : m_security(std::move(security)),
-      m_volume(0),
-      m_high_changed(false),
-      m_low_changed(false),
-      m_open_changed(false),
-      m_last_price_changed(false) {
-  m_volume_signal(m_volume);
-}
+      m_volume(0) {}
 
 void LocalTechnicalsModel::set_close(Money price) {
   m_close = price;
   m_close_signal(*m_close);
 }
 
 void LocalTechnicalsModel::update(const TimeAndSale& time_and_sale) {
-  if(!m_open.is_initialized()) {
-    m_open = time_and_sale.m_price;
-    m_open_changed = true;
-  }
-  if(time_and_sale.m_price != m_last_price) {
+  if(!m_last_price.is_initialized() || time_and_sale.m_price != m_last_price) {
```

Comment:
Use the following reference to eliminate checks to ```is_initialized```:

https://www.boost.org/doc/libs/1_69_0/libs/optional/doc/html/boost_optional/tutorial/relational_operators.html